pkg/kubelet/cm/dra/manager.go

```diff
@@ -253,3 +232,26 @@ func (m *ManagerImpl) UnprepareResources(pod *v1.Pod) error {
 func (m *ManagerImpl) PodMightNeedToUnprepareResources(UID types.UID) bool {
 	return m.cache.hasPodReference(UID)
 }
+
+// GetPodClaimInfo gets Pod's ClaimInfo List
+func (m *ManagerImpl) GetPodClaimInfoList(pod *v1.Pod, container *v1.Container) ([]*claimInfo, error) {
```

Comment:
This comment doesn't match the name of the function. Also, I'd suggest just calling it `GetPodClaimInfos()`.

Also, it is not recommended to return an unexprted type like this (i.e. `claimInfo` as lower-case). We should consider exporting it if we are going to return it out of this package like this.

Reply:
Or maybe the better thing is just to transform this into a function that returns a `ClaimInfoStateList` (which can then be used internally as well when we generate this list in `syncToCheckpoint()`).

Reply:
I am not sure how we can reuse it. Here we want to return the claimInfoState per Pod and in the syncToCheckpoint we don't care about what claimInfo belongs to per pod. (we just want to checkpoint all of them)

I can do as you suggested to return the ClaimInfoStateList  instead of the claimInfo itself.